pandas/io/parsers/python_parser.py

```diff
@@ -154,12 +157,7 @@ def __init__(self, f: ReadCsvBuffer[str] | list, **kwds) -> None:
             self._col_indices = list(range(len(self.columns)))
 
         self._parse_date_cols = self._validate_parse_dates_presence(self.columns)
-        no_thousands_columns: set[int] | None = None
-        if self.parse_dates:
-            no_thousands_columns = self._set_noconvert_dtype_columns(
-                self._col_indices, self.columns
-            )
-        self._no_thousands_columns = no_thousands_columns
+        self._no_thousands_columns = self._set_no_thousand_columns()
```

Comment:
it looks like now, `self._no_thousands_columns` will always be a set?

if so, in

https://github.com/pandas-dev/pandas/blob/b4708bd1c8a6b6dc1876cbaa193b19492ad33055/pandas/io/parsers/python_parser.py#L892

is the first check redundant? Could it just be
```
i in self._no_thousands_columns
```
?

Reply:
Thank you for helping me. I agree, there is no need to check `self._no_thousands_columns`. I corrected that.